app/services/github_authenticator.rb

```diff
@@ -0,0 +1,68 @@
+require 'httparty'
+
+# Handles Github authentication
+#
+# @!visibility private
+class GithubAuthenticator < BaseAuthenticator
+
+  PROVIDER = 'github'.freeze
+  TOKEN_URL = 'https://github.com/login/oauth/access_token'.freeze
+  USER_URL = 'https://api.github.com/user?access_token=%{access_token}'.freeze
+
+  def initialize(auth_state, auth_code)
+    @auth_code = auth_code
+    @auth_state = auth_state
+    @access_token = nil
+  end
+
+  def github_token
+    access_token
+  end
+
+  private
+
+    def connect_login_to_account(login, user)
+      login.update_attributes!(uid: user[:id], provider: PROVIDER)
+    end
+
+    def create_login_from_account(user)
+      login_attributes = {
+        identification: user[:email],
+        uid: user[:id],
+        provider: PROVIDER
+      }
+
+      Login.create!(login_attributes)
```

Comment:
I think once we ever get around to extract the ˋIdentityˋ model from the ˋLoginˋ model (see #15), that new model would be the ideal place to persist the github auth token so it could be used (and potentially be exposed to the client) later.